### PR TITLE
Native sql timeout upgrade guide

### PR DESCRIPTION
**Pull request by SzymonPobiega**
To be merged after https://github.com/Particular/docs.particular.net/pull/2815

- [x] Add link in transports/sql/native-delayed-delivery.md

### CODE REVIEW

File: samples/sqltransport/native-timeout-migration/SqlServer_3.1/Endpoint.Native/MigrateFromNHibernate.sql [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+declare @dalayedTableSuffix nvarchar(max) = 'Delayed';
+
+declare @migrateScript nvarchar(max);
+set @migrateScript = '
+    WITH message AS (
+		SELECT * 
+		FROM TimeoutEntity WITH (UPDLOCK, READPAST, ROWLOCK) 
+		WHERE [Endpoint] = ''' + @endpointName + '''
+			AND [Time] IS NOT NULL)
+	DELETE FROM message
```
Comment (MarcinHoppe) [inline samples/sqltransport/native-timeout-migration/SqlServer_3.1/Endpoint.Native/MigrateFromNHibernate.sql]: Smells like unbounded delete. Maybe we should add a note that if number of scheduled timeouts is significant, then add batching.

Or maybe turn this into batched deletes right from the start?

File: samples/sqltransport/native-timeout-migration/SqlServer_3.1/Endpoint.Native/MigrateFromSql.sql [hunk 2]
```diff
@@ -0,0 +1,24 @@
+﻿-- startcode MigrateFromSql
+
+declare @endpointName nvarchar(max) = 'PUT ENDPOINT NAME HERE';
+
+declare @endpointSchema nvarchar(max) = 'dbo';
+declare @dalayedTableSuffix nvarchar(max) = 'Delayed';
+declare @timeoutDataTable nvarchar(max) = REPLACE(@endpointName,'.','_') + '_TimeoutData';
+
+declare @migrateScript nvarchar(max);
+set @migrateScript = '
```
Comment (MarcinHoppe) [inline samples/sqltransport/native-timeout-migration/SqlServer_3.1/Endpoint.Native/MigrateFromSql.sql]: When concatenating `nvarchar`s, it's probably a good idea to prefix all the literals with `N`:

`set @migrateScript = N'`

**Comment by weralabaj** [comment]
rebased

File: samples/sqltransport/native-timeout-migration/SqlServer_3.1/Endpoint.Native/MigrateFromNHibernate.sql (hunk 1 above)
Comment (weralabaj) [inline samples/sqltransport/native-timeout-migration/SqlServer_3.1/Endpoint.Native/MigrateFromNHibernate.sql]: I'd add a note, adding a batch here would complicate the code.

Comment (MarcinHoppe) [inline samples/sqltransport/native-timeout-migration/SqlServer_3.1/Endpoint.Native/MigrateFromNHibernate.sql]: Agreed.

**Comment by weralabaj** [comment]
@MarcinHoppe  @SzymonPobiega please see if it's ok now, I'll do a proper review now  too:)

File: samples/sqltransport/native-timeout-migration/SqlTransport_3.1/Endpoint.Native/Properties/AssemblyInfo.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
+﻿using System.Reflection;
```
Comment (SzymonPobiega) [inline samples/sqltransport/native-timeout-migration/SqlTransport_3.1/Endpoint.Native/Properties/AssemblyInfo.cs]: 🔥 this file.

**Comment by SzymonPobiega** [comment]
I've rebased it. Should built fine now.

**Comment by SzymonPobiega** [comment]
It is green 🎉 . @weralabaj I assume you're done and we're waiting for @MarcinHoppe

**Comment by weralabaj** [comment]
yes, correct

**Comment by MarcinHoppe** [comment]
@weralabaj @SzymonPobiega On it now.

File: samples/sqltransport/native-timeout-migration/sample.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+
+## Running the project
+
+ 1. Start the Endpoint.NHibernate and Endpoint.SqlPersistence projects.
+ 1. Close them after `The timeout has been requested. Press any key to exit` is displayed.
+ 1. Open `MigrateFromNHibernate.sql` script in SQL Server Management Studio.
+ 1. Replace the endpoint name value in the script with `Samples.SqlServer.NativeTimeoutMigration`
+ 1. Run the script. The outcome should be `(1 row(s) affected)`.
+ 1. Repeat the steps 3-5 for the `MigrateFromNHibernate.sql` script.
```
Comment (MarcinHoppe) [inline samples/sqltransport/native-timeout-migration/sample.md]: The NHibernate script again? Or should it be MigrateFromSql.sql?

Comment (SzymonPobiega) [inline samples/sqltransport/native-timeout-migration/sample.md]: Yeah, a copy-paste error. Fixed.

**Pull request merged**